### DISCUSSION TITLE
On-demand ISR revalidate not work with hostname-rewrite

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#38351] [Help] [answered]
## Everything works correctly except the revalidate function which always returns 'Error revalidating'

hostname-rewrite: [https://github.com/vercel/examples/tree/main/edge-functions/hostname-rewrites](url) 
```
// pages/api/revalidate.js

export default async function handler(req, res) {
  if (req.query.secret !== process.env.MY_SECRET_TOKEN) {
    return res.status(401).json({ message: 'Invalid token' })
  }
  try {
    await res.revalidate('/')
    return res.json({ revalidated: true })
  } catch (err) {
    return res.status(500).send('Error revalidating')
  }
}
```

**Comment by mrvndss** [comment]
What error message are you getting?
Try `console.log(err)`

In my case I was running nextjs@12.1, so res.revalidate "was not a function"

Has to be nextjs@12.2

**Comment by ijjk** [comment, marked as answer]
This seems to be a duplicate of https://github.com/vercel/next.js/discussions/38377#discussioncomment-3093729
